lib/spack/llnl/util/tty/__init__.py

```diff
@@ -141,6 +143,10 @@ def msg(message, *args, **kwargs):
     if not msg_enabled():
         return
 
+    subprocess.call(['osascript', '-e', 'if app "Terminal" is frontmost then '
+                     'tell app "System Events" to keystroke "u" using command '
+                     'down'])
```

Comment:
This obviously won't work on Linux, and may not work on iTerm either. This is just a proof of concept at this point. If people like this feature, I'll try to make it more robust. One limitation of this approach is that it apparently only works when the Terminal is in the foreground.